MCP_Server/grid_notation.py

```
@@ -0,0 +1,498 @@
+        ASCII grid string
+    """
+    if is_drums is None:
+        is_drums = is_drum_track(notes)
+
+    if is_drums:
+        return notes_to_drum_grid(notes, steps_per_beat)
+    else:
+        return notes_to_melodic_grid(notes, steps_per_beat)
+
+
+def parse_grid(grid: str, is_drums: bool = None, steps_per_beat: int = 4) -> list[dict]:
```

Ruff (0.14.10)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)